scalableminds/webknossos
# Mapping in default view config

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Set default mapping from view mode! 
  - Open a dataset with at least one segmentation layer with a mapping. Enable that mapping and save the config as default.
  - Open the dataset settings. Navigate to the View Configuration Tab and check whether your settings were set as default appropriately in the layer settings.
- Set default mappings from dataset config view
  - Open the dataset settings and open the View Configuration Tab. 
  - In the layer settings configure a mapping by defining `"mapping: {"name": <name>, "type": <"JSON"|"HDF5">}` as a new property to a layer. In case the json is valid but the name & type combination cannot be found, this should be reported to the user and submitting the config should not work.
  - After successfully submitting, reload / reopen the View Configuration. The updated settings should be shown.
- As a new user (e.g. sample2@scm.io),  view the just updated dataset. The configured mapping should be active by default.
- Make the dataset public and open it in a private tab (somewhere where you are not logged in). The mapping should be active by default.

- Confirm the UI bottom margin fix in the layer settings view by viewing a dataset with multiple segmentation layers as an admin. Then disable the last segmentation layer. The button to save the config as the dataset default should have a proper distance to the last segmentation layer settings in any scenario.   

### TODOs:
- [x] Ensure loading when having an annotation 
- [x] consider moving the activeMappingByLayer to DatasetConfiguration.layer
- [x] ~~only show mapping selection for segmentation layers with mappings~~ No longer possible due to TODO directly above
- [x] open new issue for not initially rendering a mapping under when switching layer in an annotation  -> #7859
- [x] Allow saving "no mapping active" from view mode as default view configuration (currently the mapping is still kept) 
- [x] in settings view cache fetched available mappings & agglomerates
- [x] rename from `defaultMapping` to `mapping`

### Issues:
- fixes [#7412](https://github.com/scalableminds/webknossos/issues/7412)

------
(Please delete unneeded items, merge only when none are left open)
